// vue-sonner/lib/vue-sonner.js
var xt = Object.defineProperty;
var Tt = (i, e, t) => e in i ? xt(i, e, { enumerable: !0, configurable: !0, writable: !0, value: t }) : i[e] = t;
var w = (i, e, t) => (Tt(i, typeof e != "symbol" ? e + "" : e, t), t);
import { defineComponent as dt, openBlock as d, createElementBlock as g, createElementVNode as P, Fragment as U, renderList as rt, unref as V, ref as f, watchEffect as D, computed as h, onMounted as Bt, onUnmounted as $t, normalizeClass as nt, normalizeStyle as it, createVNode as Ct, createCommentVNode as M, createBlock as H, createTextVNode as ft, toDisplayString as j, resolveDynamicComponent as St, useAttrs as Nt, nextTick as Et, watch as pt } from "vue";
function kt(i) {
  if (!i || typeof document > "u")
    return;
  let e = document.head || document.getElementsByTagName("head")[0], t = document.createElement("style");
  t.type = "text/css", e.appendChild(t), t.styleSheet ? t.styleSheet.cssText = i : t.appendChild(document.createTextNode(i));
}
kt("[data-sonner-toaster][dir=ltr],html[dir=ltr]{--toast-icon-margin-start:-3px;--toast-icon-margin-end:4px;--toast-svg-margin-start:-1px;--toast-svg-margin-end:0px;--toast-button-margin-start:auto;--toast-button-margin-end:0;--toast-close-button-start:0;--toast-close-button-end:unset;--toast-close-button-transform:translate(-35%, -35%)}[data-sonner-toaster][dir=rtl],html[dir=rtl]{--toast-icon-margin-start:4px;--toast-icon-margin-end:-3px;--toast-svg-margin-start:0px;--toast-svg-margin-end:-1px;--toast-button-margin-start:0;--toast-button-margin-end:auto;--toast-close-button-start:unset;--toast-close-button-end:0;--toast-close-button-transform:translate(35%, -35%)}[data-sonner-toaster]{position:fixed;width:var(--width);font-family:ui-sans-serif,system-ui,-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica Neue,Arial,Noto Sans,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol,Noto Color Emoji;--gray1:hsl(0, 0%, 99%);--gray2:hsl(0, 0%, 97.3%);--gray3:hsl(0, 0%, 95.1%);--gray4:hsl(0, 0%, 93%);--gray5:hsl(0, 0%, 90.9%);--gray6:hsl(0, 0%, 88.7%);--gray7:hsl(0, 0%, 85.8%);--gray8:hsl(0, 0%, 78%);--gray9:hsl(0, 0%, 56.1%);--gray10:hsl(0, 0%, 52.3%);--gray11:hsl(0, 0%, 43.5%);--gray12:hsl(0, 0%, 9%);--border-radius:8px;box-sizing:border-box;padding:0;margin:0;list-style:none;outline:0;z-index:999999999}[data-sonner-toaster][data-x-position=right]{right:max(var(--offset),env(safe-area-inset-right))}[data-sonner-toaster][data-x-position=left]{left:max(var(--offset),env(safe-area-inset-left))}[data-sonner-toaster][data-x-position=center]{left:50%;transform:translateX(-50%)}[data-sonner-toaster][data-y-position=top]{top:max(var(--offset),env(safe-area-inset-top))}[data-sonner-toaster][data-y-position=bottom]{bottom:max(var(--offset),env(safe-area-inset-bottom))}[data-sonner-toast]{--y:translateY(100%);--lift-amount:calc(var(--lift) * var(--gap));z-index:var(--z-index);position:absolute;opacity:0;transform:var(--y);touch-action:none;will-change:transform,opacity,height;transition:transform .4s,opacity .4s,height .4s,box-shadow .2s;box-sizing:border-box;outline:0;overflow-wrap:anywhere}[data-sonner-toast][data-styled=true]{padding:16px;background:var(--normal-bg);border:1px solid var(--normal-border);color:var(--normal-text);border-radius:var(--border-radius);box-shadow:0 4px 12px rgba(0,0,0,.1);width:var(--width);font-size:13px;display:flex;align-items:center;gap:6px}[data-sonner-toast]:focus-visible{box-shadow:0 4px 12px rgba(0,0,0,.1),0 0 0 2px rgba(0,0,0,.2)}[data-sonner-toast][data-y-position=top]{top:0;--y:translateY(-100%);--lift:1;--lift-amount:calc(1 * var(--gap))}[data-sonner-toast][data-y-position=bottom]{bottom:0;--y:translateY(100%);--lift:-1;--lift-amount:calc(var(--lift) * var(--gap))}[data-sonner-toast] [data-description]{font-weight:400;line-height:1.4;color:inherit}[data-sonner-toast] [data-title]{font-weight:500;line-height:1.5;color:inherit}[data-sonner-toast] [data-icon]{display:flex;height:16px;width:16px;position:relative;justify-content:flex-start;align-items:center;flex-shrink:0;margin-left:var(--toast-icon-margin-start);margin-right:var(--toast-icon-margin-end)}[data-sonner-toast][data-promise=true] [data-icon]>svg{opacity:0;transform:scale(.8);transform-origin:center;animation:sonner-fade-in .3s ease forwards}[data-sonner-toast] [data-icon]>*{flex-shrink:0}[data-sonner-toast] [data-icon] svg{margin-left:var(--toast-svg-margin-start);margin-right:var(--toast-svg-margin-end)}[data-sonner-toast] [data-content]{display:flex;flex-direction:column;gap:2px;transform:translateZ(0)}[data-sonner-toast] [data-button]{border-radius:4px;padding-left:8px;padding-right:8px;height:24px;font-size:12px;color:var(--normal-bg);background:var(--normal-text);margin-left:var(--toast-button-margin-start);margin-right:var(--toast-button-margin-end);border:none;cursor:pointer;outline:0;display:flex;align-items:center;flex-shrink:0;transition:opacity .4s,box-shadow .2s}[data-sonner-toast] [data-button]:focus-visible{box-shadow:0 0 0 2px rgba(0,0,0,.4)}[data-sonner-toast] [data-button]:first-of-type{margin-left:var(--toast-button-margin-start);margin-right:var(--toast-button-margin-end)}[data-sonner-toast] [data-cancel]{color:var(--normal-text);background:rgba(0,0,0,.08)}[data-sonner-toast][data-theme=dark] [data-cancel]{background:rgba(255,255,255,.3)}[data-sonner-toast] [data-close-button]{position:absolute;left:var(--toast-close-button-start);right:var(--toast-close-button-end);top:0;height:20px;width:20px;display:flex;justify-content:center;align-items:center;padding:0;background:var(--gray1);color:var(--gray12);border:1px solid var(--gray4);transform:var(--toast-close-button-transform);border-radius:50%;cursor:pointer;z-index:1;transition:opacity .1s,background .2s,border-color .2s}[data-sonner-toast] [data-close-button]:focus-visible{box-shadow:0 4px 12px rgba(0,0,0,.1),0 0 0 2px rgba(0,0,0,.2)}[data-sonner-toast] [data-disabled=true]{cursor:not-allowed}[data-sonner-toast]:hover [data-close-button]:hover{background:var(--gray2);border-color:var(--gray5)}[data-sonner-toast][data-swiping=true]:before{content:'';position:absolute;left:0;right:0;height:100%;z-index:-1}[data-sonner-toast][data-y-position=top][data-swiping=true]:before{bottom:50%;transform:scaleY(3) translateY(50%)}[data-sonner-toast][data-y-position=bottom][data-swiping=true]:before{top:50%;transform:scaleY(3) translateY(-50%)}[data-sonner-toast][data-swiping=false][data-removed=true]:before{content:'';position:absolute;inset:0;transform:scaleY(2)}[data-sonner-toast]:after{content:'';position:absolute;left:0;height:calc(var(--gap) + 1px);bottom:100%;width:100%}[data-sonner-toast][data-mounted=true]{--y:translateY(0);opacity:1}[data-sonner-toast][data-expanded=false][data-front=false]{--scale:var(--toasts-before) * 0.05 + 1;--y:translateY(calc(var(--lift-amount) * var(--toasts-before))) scale(calc(-1 * var(--scale)));height:var(--front-toast-height)}[data-sonner-toast]>*{transition:opacity .4s}[data-sonner-toast][data-expanded=false][data-front=false][data-styled=true]>*{opacity:0}[data-sonner-toast][data-visible=false]{opacity:0;pointer-events:none}[data-sonner-toast][data-mounted=true][data-expanded=true]{--y:translateY(calc(var(--lift) * var(--offset)));height:var(--initial-height)}[data-sonner-toast][data-removed=true][data-front=true][data-swipe-out=false]{--y:translateY(calc(var(--lift) * -100%));opacity:0}[data-sonner-toast][data-removed=true][data-front=false][data-swipe-out=false][data-expanded=true]{--y:translateY(calc(var(--lift) * var(--offset) + var(--lift) * -100%));opacity:0}[data-sonner-toast][data-removed=true][data-front=false][data-swipe-out=false][data-expanded=false]{--y:translateY(40%);opacity:0;transition:transform .5s,opacity .2s}[data-sonner-toast][data-removed=true][data-front=false]:before{height:calc(var(--initial-height) + 20%)}[data-sonner-toast][data-swiping=true]{transform:var(--y) translateY(var(--swipe-amount,0));transition:none}[data-sonner-toast][data-swipe-out=true][data-y-position=bottom],[data-sonner-toast][data-swipe-out=true][data-y-position=top]{animation:swipe-out .2s ease-out forwards}@keyframes swipe-out{from{transform:translateY(calc(var(--lift) * var(--offset) + var(--swipe-amount)));opacity:1}to{transform:translateY(calc(var(--lift) * var(--offset) + var(--swipe-amount) + var(--lift) * -100%));opacity:0}}@media (max-width:600px){[data-sonner-toaster]{position:fixed;--mobile-offset:16px;right:var(--mobile-offset);left:var(--mobile-offset);width:100%}[data-sonner-toaster] [data-sonner-toast]{left:0;right:0;width:calc(100% - 32px)}[data-sonner-toaster][data-x-position=left]{left:var(--mobile-offset)}[data-sonner-toaster][data-y-position=bottom]{bottom:20px}[data-sonner-toaster][data-y-position=top]{top:20px}[data-sonner-toaster][data-x-position=center]{left:var(--mobile-offset);right:var(--mobile-offset);transform:none}}[data-sonner-toaster][data-theme=light]{--normal-bg:#fff;--normal-border:var(--gray4);--normal-text:var(--gray12);--success-bg:hsl(143, 85%, 96%);--success-border:hsl(145, 92%, 91%);--success-text:hsl(140, 100%, 27%);--info-bg:hsl(208, 100%, 97%);--info-border:hsl(221, 91%, 91%);--info-text:hsl(210, 92%, 45%);--warning-bg:hsl(49, 100%, 97%);--warning-border:hsl(49, 91%, 91%);--warning-text:hsl(31, 92%, 45%);--error-bg:hsl(359, 100%, 97%);--error-border:hsl(359, 100%, 94%);--error-text:hsl(360, 100%, 45%)}[data-sonner-toaster][data-theme=light] [data-sonner-toast][data-invert=true]{--normal-bg:#000;--normal-border:hsl(0, 0%, 20%);--normal-text:var(--gray1)}[data-sonner-toaster][data-theme=dark] [data-sonner-toast][data-invert=true]{--normal-bg:#fff;--normal-border:var(--gray3);--normal-text:var(--gray12)}[data-sonner-toaster][data-theme=dark]{--normal-bg:#000;--normal-border:hsl(0, 0%, 20%);--normal-text:var(--gray1);--success-bg:hsl(150, 100%, 6%);--success-border:hsl(147, 100%, 12%);--success-text:hsl(150, 86%, 65%);--info-bg:hsl(215, 100%, 6%);--info-border:hsl(223, 100%, 12%);--info-text:hsl(216, 87%, 65%);--warning-bg:hsl(64, 100%, 6%);--warning-border:hsl(60, 100%, 12%);--warning-text:hsl(46, 87%, 65%);--error-bg:hsl(358, 76%, 10%);--error-border:hsl(357, 89%, 16%);--error-text:hsl(358, 100%, 81%)}[data-rich-colors=true] [data-sonner-toast][data-type=success]{background:var(--success-bg);border-color:var(--success-border);color:var(--success-text)}[data-rich-colors=true] [data-sonner-toast][data-type=success] [data-close-button]{background:var(--success-bg);border-color:var(--success-border);color:var(--success-text)}[data-rich-colors=true] [data-sonner-toast][data-type=info]{background:var(--info-bg);border-color:var(--info-border);color:var(--info-text)}[data-rich-colors=true] [data-sonner-toast][data-type=info] [data-close-button]{background:var(--info-bg);border-color:var(--info-border);color:var(--info-text)}[data-rich-colors=true] [data-sonner-toast][data-type=warning]{background:var(--warning-bg);border-color:var(--warning-border);color:var(--warning-text)}[data-rich-colors=true] [data-sonner-toast][data-type=warning] [data-close-button]{background:var(--warning-bg);border-color:var(--warning-border);color:var(--warning-text)}[data-rich-colors=true] [data-sonner-toast][data-type=error]{background:var(--error-bg);border-color:var(--error-border);color:var(--error-text)}[data-rich-colors=true] [data-sonner-toast][data-type=error] [data-close-button]{background:var(--error-bg);border-color:var(--error-border);color:var(--error-text)}.sonner-loading-wrapper{--size:16px;height:var(--size);width:var(--size);position:absolute;inset:0;z-index:10}.sonner-loading-wrapper[data-visible=false]{transform-origin:center;animation:sonner-fade-out .2s ease forwards}.sonner-spinner{position:relative;top:50%;left:50%;height:var(--size);width:var(--size)}.sonner-loading-bar{animation:sonner-spin 1.2s linear infinite;background:var(--gray11);border-radius:6px;height:8%;left:-10%;position:absolute;top:-3.9%;width:24%}.sonner-loading-bar:first-child{animation-delay:-1.2s;transform:rotate(.0001deg) translate(146%)}.sonner-loading-bar:nth-child(2){animation-delay:-1.1s;transform:rotate(30deg) translate(146%)}.sonner-loading-bar:nth-child(3){animation-delay:-1s;transform:rotate(60deg) translate(146%)}.sonner-loading-bar:nth-child(4){animation-delay:-.9s;transform:rotate(90deg) translate(146%)}.sonner-loading-bar:nth-child(5){animation-delay:-.8s;transform:rotate(120deg) translate(146%)}.sonner-loading-bar:nth-child(6){animation-delay:-.7s;transform:rotate(150deg) translate(146%)}.sonner-loading-bar:nth-child(7){animation-delay:-.6s;transform:rotate(180deg) translate(146%)}.sonner-loading-bar:nth-child(8){animation-delay:-.5s;transform:rotate(210deg) translate(146%)}.sonner-loading-bar:nth-child(9){animation-delay:-.4s;transform:rotate(240deg) translate(146%)}.sonner-loading-bar:nth-child(10){animation-delay:-.3s;transform:rotate(270deg) translate(146%)}.sonner-loading-bar:nth-child(11){animation-delay:-.2s;transform:rotate(300deg) translate(146%)}.sonner-loading-bar:nth-child(12){animation-delay:-.1s;transform:rotate(330deg) translate(146%)}@keyframes sonner-fade-in{0%{opacity:0;transform:scale(.8)}100%{opacity:1;transform:scale(1)}}@keyframes sonner-fade-out{0%{opacity:1;transform:scale(1)}100%{opacity:0;transform:scale(.8)}}@keyframes sonner-spin{0%{opacity:1}100%{opacity:.15}}@media (prefers-reduced-motion){.sonner-loading-bar,[data-sonner-toast],[data-sonner-toast]>*{transition:none!important;animation:none!important}}.sonner-loader{position:absolute;top:50%;left:50%;transform:translate(-50%,-50%);transform-origin:center;transition:opacity .2s,transform .2s}.sonner-loader[data-visible=false]{opacity:0;transform:scale(.8) translate(-50%,-50%)}");
let lt = 1;
class It {
  constructor() {
    w(this, "subscribers");
    w(this, "toasts");
    // We use arrow functions to maintain the correct `this` reference
    w(this, "subscribe", (e) => (this.subscribers.push(e), () => {
      const t = this.subscribers.indexOf(e);
      this.subscribers.splice(t, 1);
    }));
    w(this, "publish", (e) => {
      this.subscribers.forEach((t) => t(e));
    });
    w(this, "addToast", (e) => {
      this.publish(e), this.toasts = [...this.toasts, e];
    });
    w(this, "create", (e) => {
      var y;
      const { message: t, ...n } = e, o = typeof (e == null ? void 0 : e.id) == "number" || ((y = e.id) == null ? void 0 : y.length) > 0 ? e.id : lt++, u = this.toasts.findIndex((c) => c.id === o), p = e.dismissible === void 0 ? !0 : e.dismissible;
      return u !== -1 ? this.toasts = this.toasts.map((c) => c.id === o ? (this.publish({ ...c, ...e, id: o, title: t }), {
        ...c,
        ...e,
        id: o,
        dismissible: p,
        title: t
      }) : c) : this.addToast({ title: t, ...n, dismissible: p, id: o }), o;
    });
    w(this, "dismiss", (e) => (e || this.toasts.forEach((t) => {
      this.subscribers.forEach(
        (n) => n({ id: t.id, dismiss: !0 })
      );
    }), this.subscribers.forEach((t) => t({ id: e, dismiss: !0 })), e));
    w(this, "message", (e, t) => this.create({ ...t, message: e }));
    w(this, "error", (e, t) => this.create({ ...t, type: "error", message: e }));
    w(this, "success", (e, t) => this.create({ ...t, type: "success", message: e }));
    w(this, "info", (e, t) => this.create({ ...t, type: "info", message: e }));
    w(this, "warning", (e, t) => this.create({ ...t, type: "warning", message: e }));
    w(this, "loading", (e, t) => this.create({ ...t, type: "loading", message: e }));
    w(this, "promise", (e, t) => {
      if (!t)
        return;
      let n;
      t.loading !== void 0 && (n = this.create({
        ...t,
        promise: e,
        type: "loading",
        message: t.loading,
        description: typeof t.description != "function" ? t.description : void 0
      }));
      const o = e instanceof Promise ? e : e();
      let u = n !== void 0;
      return o.then((p) => {
        if (p && // @ts-expect-error
        typeof p.ok == "boolean" && // @ts-expect-error
        !p.ok) {
          u = !1;
          const y = typeof t.error == "function" ? (
            // @ts-expect-error
            t.error(`HTTP error! status: ${response.status}`)
          ) : t.error, c = typeof t.description == "function" ? (
            // @ts-expect-error
            t.description(`HTTP error! status: ${response.status}`)
          ) : t.description;
          this.create({ id: n, type: "error", message: y, description: c });
        } else if (t.success !== void 0) {
          u = !1;
          const y = typeof t.success == "function" ? t.success(p) : t.success, c = typeof t.description == "function" ? (
            // @ts-expect-error
            t.description(p)
          ) : t.description;
          this.create({ id: n, type: "success", message: y, description: c });
        }
      }).catch((p) => {
        if (t.error !== void 0) {
          u = !1;
          const y = typeof t.error == "function" ? t.error(p) : t.error, c = typeof t.description == "function" ? (
            // @ts-expect-error
            t.description(p)
          ) : t.description;
          this.create({ id: n, type: "error", message: y, description: c });
        }
      }).finally(() => {
        var p;
        u && (this.dismiss(n), n = void 0), (p = t.finally) == null || p.call(t);
      }), n;
    });
    // We can't provide the toast we just created as a prop as we didn't create it yet, so we can create a default toast object, I just don't know how to use function in argument when calling()?
    w(this, "custom", (e, t) => {
      const n = (t == null ? void 0 : t.id) || lt++;
      return this.publish({ ...t, id: n, title: e }), n;
    });
    this.subscribers = [], this.toasts = [];
  }
}
const S = new It(), _t = (i, e) => {
  const t = (e == null ? void 0 : e.id) || lt++;
  return S.addToast({
    title: i,
    ...e,
    id: t
  }), t;
}, Pt = _t, ze = Object.assign(Pt, {
  success: S.success,
  info: S.info,
  warning: S.warning,
  error: S.error,
  custom: S.custom,
  message: S.message,
  promise: S.promise,
  dismiss: S.dismiss,
  loading: S.loading
}), zt = ["data-visible"], Mt = { className: "sonner-spinner" }, Ot = /* @__PURE__ */ dt({
  __name: "Loader",
  props: {
    visible: Boolean
  },
  setup(i) {
    const e = Array(12).fill(0);
    return (t, n) => (d(), g("div", {
      className: "sonner-loading-wrapper",
      "data-visible": i.visible
    }, [
      P("div", Mt, [
        (d(!0), g(U, null, rt(V(e), (o) => (d(), g("div", {
          key: `spinner-bar-${o}`,
          className: "sonner-loading-bar"
        }))), 128))
      ])
    ], 8, zt));
  }
}), K = (i, e) => {
  const t = i.__vccOpts || i;
  for (const [n, o] of e)
    t[n] = o;
  return t;
}, At = {}, Ht = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 20 20",
  fill: "currentColor",
  height: "20",
  width: "20"
}, Dt = /* @__PURE__ */ P("path", {
  "fill-rule": "evenodd",
  d: "M10 18a8 8 0 100-16 8 8 0 000 16zm3.857-9.809a.75.75 0 00-1.214-.882l-3.483 4.79-1.88-1.88a.75.75 0 10-1.06 1.061l2.5 2.5a.75.75 0 001.137-.089l4-5.5z",
  "clip-rule": "evenodd"
}, null, -1), Lt = [
  Dt
];
function Yt(i, e) {
  return d(), g("svg", Ht, Lt);
}
const Rt = /* @__PURE__ */ K(At, [["render", Yt]]), Ft = {}, Wt = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 20 20",
  fill: "currentColor",
  height: "20",
  width: "20"
}, jt = /* @__PURE__ */ P("path", {
  "fill-rule": "evenodd",
  d: "M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-7-4a1 1 0 11-2 0 1 1 0 012 0zM9 9a.75.75 0 000 1.5h.253a.25.25 0 01.244.304l-.459 2.066A1.75 1.75 0 0010.747 15H11a.75.75 0 000-1.5h-.253a.25.25 0 01-.244-.304l.459-2.066A1.75 1.75 0 009.253 9H9z",
  "clip-rule": "evenodd"
}, null, -1), Ut = [
  jt
];
function Vt(i, e) {
  return d(), g("svg", Wt, Ut);
}
const Kt = /* @__PURE__ */ K(Ft, [["render", Vt]]), Xt = {}, Gt = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 24 24",
  fill: "currentColor",
  height: "20",
  width: "20"
}, Zt = /* @__PURE__ */ P("path", {
  "fill-rule": "evenodd",
  d: "M9.401 3.003c1.155-2 4.043-2 5.197 0l7.355 12.748c1.154 2-.29 4.5-2.599 4.5H4.645c-2.309 0-3.752-2.5-2.598-4.5L9.4 3.003zM12 8.25a.75.75 0 01.75.75v3.75a.75.75 0 01-1.5 0V9a.75.75 0 01.75-.75zm0 8.25a.75.75 0 100-1.5.75.75 0 000 1.5z",
  "clip-rule": "evenodd"
}, null, -1), qt = [
  Zt
];
function Jt(i, e) {
  return d(), g("svg", Gt, qt);
}
const Qt = /* @__PURE__ */ K(Xt, [["render", Jt]]), te = {}, ee = {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 20 20",
  fill: "currentColor",
  height: "20",
  width: "20"
}, ae = /* @__PURE__ */ P("path", {
  "fill-rule": "evenodd",
  d: "M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-8-5a.75.75 0 01.75.75v4.5a.75.75 0 01-1.5 0v-4.5A.75.75 0 0110 5zm0 10a1 1 0 100-2 1 1 0 000 2z",
  "clip-rule": "evenodd"
}, null, -1), oe = [
  ae
];
function se(i, e) {
  return d(), g("svg", ee, oe);
}
const ne = /* @__PURE__ */ K(te, [["render", se]]), re = {}, ie = {
  xmlns: "http://www.w3.org/2000/svg",
  width: "12",
  height: "12",
  viewBox: "0 0 24 24",
  fill: "none",
  stroke: "currentColor",
  "stoke-width": "1.5",
  "stroke-linecap": "round",
  "stroke-linejoin": "round"
}, le = /* @__PURE__ */ P("line", {
  x1: "18",
  y1: "6",
  x2: "6",
  y2: "18"
}, null, -1), de = /* @__PURE__ */ P("line", {
  x1: "6",
  y1: "6",
  x2: "18",
  y2: "18"
}, null, -1), ce = [
  le,
  de
];
function ue(i, e) {
  return d(), g("svg", ie, ce);
}
const fe = /* @__PURE__ */ K(re, [["render", ue]]), pe = () => {
  const i = f(!1);
  return D(() => {
    const e = () => {
      i.value = document.hidden;
    };
    return document.addEventListener("visibilitychange", e), () => window.removeEventListener("visibilitychange", e);
  }), {
    isDocumentHidden: i
  };
}, me = ["aria-live", "data-styled", "data-mounted", "data-promise", "data-removed", "data-visible", "data-y-position", "data-x-position", "data-index", "data-front", "data-swiping", "data-dismissible", "data-type", "data-invert", "data-swipe-out", "data-expanded"], he = ["aria-label", "data-disabled"], ge = {
  key: 1,
  "data-icon": ""
}, ve = { "data-content": "" }, be = { "data-title": "" }, ye = 4e3, we = 14, xe = 20, Te = 200, ke = /* @__PURE__ */ dt({
  __name: "Toast",
  props: {
    toast: {},
    toasts: {},
    index: {},
    expanded: { type: Boolean },
    invert: { type: Boolean },
    heights: {},
    gap: {},
    position: {},
    visibleToasts: {},
    expandByDefault: { type: Boolean },
    closeButton: { type: Boolean },
    interacting: { type: Boolean },
    duration: {},
    descriptionClass: {},
    style: {},
    cancelButtonStyle: {},
    actionButtonStyle: {},
    unstyled: { type: Boolean },
    descriptionClassName: {},
    classNames: {},
    closeButtonAriaLabel: {},
    pauseWhenPageIsHidden: { type: Boolean },
    cn: { type: Function }
  },
  emits: ["update:heights", "removeToast"],
  setup(i, { emit: e }) {
    const t = (a) => !!a.promise, n = e, o = i, u = f(!1), p = f(!1), y = f(!1), c = f(!1), N = f(null), E = f(0), X = f(0), I = f(null), x = f(null), O = h(() => o.index === 0), et = h(() => o.index + 1 <= o.visibleToasts), L = h(() => o.toast.type), G = h(() => o.toast.dismissible !== !1), at = h(() => {
      var a, l, k, C, T, m, st;
      return o.cn(
        (a = o.classNames) == null ? void 0 : a.toast,
        (k = (l = o.toast) == null ? void 0 : l.classNames) == null ? void 0 : k.toast,
        (C = o.classNames) == null ? void 0 : C.default,
        (T = o.classNames) == null ? void 0 : T[o.toast.type || "default"],
        (st = (m = o.toast) == null ? void 0 : m.classNames) == null ? void 0 : st[o.toast.type || "default"]
      );
    }), ot = o.toast.descriptionClassName || "", s = o.toast.style || {}, r = h(
      () => o.heights.findIndex((a) => a.toastId === o.toast.id) || 0
    ), v = h(() => o.toast.closeButton ?? o.closeButton), B = h(
      () => o.toast.duration || o.duration || ye
    ), $ = f(0), b = f(0), z = f(B.value), W = f(0), _ = f(null), Y = h(() => o.position.split("-")), Z = h(() => Y.value[0]), q = h(() => Y.value[1]), J = h(() => o.heights.reduce((a, l, k) => k >= r.value ? a : a + l.height, 0)), Q = pe(), tt = h(() => o.toast.invert || o.invert), R = h(() => N.value === "loading"), F = h(
      () => N.value ?? (o.toast.type || null)
    ), ct = h(() => !t(o.toast) && typeof o.toast.title == "object"), gt = h(() => {
      if (!t(o.toast))
        return null;
      switch (N.value) {
        case "loading":
          return o.toast.loading;
        case "success":
          return typeof o.toast.success == "function" ? I.value : o.toast.success;
        case "error":
          return typeof o.toast.error == "function" ? I.value : o.toast.error;
        default:
          return null;
      }
    }), vt = () => {
      var a, l;
      (!R.value || G.value) && (A(), (l = (a = o.toast).onDismiss) == null || l.call(a, o.toast));
    }, A = () => {
      p.value = !0, E.value = b.value;
      const a = o.heights.filter(
        (l) => l.toastId !== o.toast.id
      );
      n("update:heights", a), setTimeout(() => {
        n("removeToast", o.toast);
      }, Te);
    }, bt = (a) => {
      R || (E.value = b.value, a.target.setPointerCapture(a.pointerId), a.target.tagName !== "BUTTON" && (y.value = !0, _.value = { x: a.clientX, y: a.clientY }));
    }, yt = (a) => {
      var k, C, T, m;
      if (c.value)
        return;
      _.value = null;
      const l = Number(
        ((k = x.value) == null ? void 0 : k.style.getPropertyValue("--swipe-amount").replace("px", "")) || 0
      );
      if (Math.abs(l) >= xe) {
        E.value = b.value, (T = (C = o.toast).onDismiss) == null || T.call(C, o.toast), A(), c.value = !0;
        return;
      }
      (m = x.value) == null || m.style.setProperty("--swipe-amount", "0px"), y.value = !0;
    }, wt = (a) => {
      var ut;
      if (!_.value)
        return;
      const l = a.clientY - _.value.y, k = a.clientX - _.value.x, T = (Y.value[0] === "top" ? Math.min : Math.max)(0, l), m = a.pointerType === "touch" ? 10 : 2;
      Math.abs(T) > m ? (ut = x.value) == null || ut.style.setProperty("--swipe-amount", `${l}px`) : Math.abs(k) > m && (_.value = null);
    };
    return D(() => {
      b.value = r.value * we + J.value;
    }), D((a) => {
      if (o.toast.promise && N.value === "loading" || o.toast.duration === 1 / 0)
        return;
      let l;
      const k = () => {
        if (W.value < $.value) {
          const T = (/* @__PURE__ */ new Date()).getTime() - $.value;
          z.value = z.value - T;
        }
        W.value = (/* @__PURE__ */ new Date()).getTime();
      }, C = () => {
        $.value = (/* @__PURE__ */ new Date()).getTime(), l = setTimeout(() => {
          var T, m;
          (m = (T = o.toast).onAutoClose) == null || m.call(T, o.toast), A();
        }, z.value);
      };
      o.expanded || o.interacting || o.pauseWhenPageIsHidden && Q ? k() : C(), a(() => {
        clearTimeout(l);
      });
    }), D(() => {
      o.toast.delete && A();
    }), Bt(() => {
      if (x.value) {
        const a = x.value.getBoundingClientRect().height;
        X.value = a;
        const l = [
          { toastId: o.toast.id, height: a, position: o.toast.position },
          ...o.heights
        ];
        n("update:heights", l);
      }
      u.value = !0;
    }), $t(() => {
      if (x.value) {
        const a = o.heights.filter(
          (l) => l.toastId !== o.toast.id
        );
        n("update:heights", a);
      }
    }), (a, l) => {
      var k, C, T;
      return d(), g("li", {
        "aria-live": a.toast.important ? "assertive" : "polite",
        "aria-atomic": "",
        role: "status",
        tabindex: "0",
        ref_key: "toastRef",
        ref: x,
        "data-sonner-toast": "",
        class: nt(at.value),
        "data-styled": !(a.toast.unstyled || a.unstyled),
        "data-mounted": u.value,
        "data-promise": !!a.toast.promise,
        "data-removed": p.value,
        "data-visible": et.value,
        "data-y-position": Z.value,
        "data-x-position": q.value,
        "data-index": a.index,
        "data-front": O.value,
        "data-swiping": y.value,
        "data-dismissible": G.value,
        "data-type": L.value,
        "data-invert": tt.value,
        "data-swipe-out": c.value,
        "data-expanded": !!(a.expanded || a.expandByDefault && u.value),
        style: it({
          "--index": a.index,
          "--toasts-before": a.index,
          "--z-index": a.toasts.length - a.index,
          "--offset": `${p.value ? E.value : b.value}px`,
          "--initial-height": a.expandByDefault ? "auto" : `${X.value}px`,
          ...a.style,
          ...V(s)
        }),
        onPointerdown: bt,
        onPointerup: yt,
        onPointermove: wt
      }, [
        v.value && !ct.value ? (d(), g("button", {
          key: 0,
          "aria-label": a.closeButtonAriaLabel || "Close toast",
          "data-disabled": R.value,
          "data-close-button": "",
          class: nt(a.cn((k = a.classNames) == null ? void 0 : k.closeButton, (T = (C = a.toast) == null ? void 0 : C.classNames) == null ? void 0 : T.closeButton)),
          onClick: vt
        }, [
          Ct(fe)
        ], 10, he)) : M("", !0),
        L.value || a.toast.icon || a.toast.promise ? (d(), g("div", ge, [
          typeof a.toast.promise == "function" || L.value === "loading" ? (d(), H(Ot, {
            key: 0,
            visible: N.value === "loading" || L.value === "loading"
          }, null, 8, ["visible"])) : M("", !0),
          F.value === "success" ? (d(), H(Rt, { key: 1 })) : F.value === "info" ? (d(), H(Kt, { key: 2 })) : F.value === "warning" ? (d(), H(Qt, { key: 3 })) : F.value === "error" ? (d(), H(ne, { key: 4 })) : M("", !0)
        ])) : M("", !0),
        P("div", ve, [
          P("div", be, [
            typeof a.toast.title == "string" ? (d(), g(U, { key: 0 }, [
              ft(j(a.toast.title), 1)
            ], 64)) : a.toast.title === void 0 || a.toast.title === null ? (d(), g(U, { key: 1 }, [
              ft(j(gt.value), 1)
            ], 64)) : ct.value ? (d(), H(St(a.toast.title), {
              key: 2,
              onCloseToast: l[0] || (l[0] = () => {
                var m;
                A(), (m = a.toast.cancel) != null && m.onClick && a.toast.cancel.onClick();
              })
            }, null, 32)) : M("", !0)
          ]),
          a.toast.description ? (d(), g("div", {
            key: 0,
            "data-description": "",
            class: nt(a.descriptionClass + V(ot))
          }, j(a.toast.description), 3)) : M("", !0)
        ]),
        a.toast.cancel ? (d(), g("button", {
          key: 2,
          "data-button": "",
          "data-cancel": "",
          onClick: l[1] || (l[1] = () => {
            var m;
            A(), (m = a.toast.cancel) != null && m.onClick && a.toast.cancel.onClick();
          })
        }, j(a.toast.cancel.label), 1)) : M("", !0),
        a.toast.action ? (d(), g("button", {
          key: 3,
          "data-button": "",
          onClick: l[2] || (l[2] = () => {
            var m;
            A(), (m = a.toast.action) == null || m.onClick();
          })
        }, j(a.toast.action.label), 1)) : M("", !0)
      ], 46, me);
    };
  }
}), Be = ["aria-label"], $e = ["dir", "data-theme", "data-rich-colors", "data-y-position", "data-x-position"], Ce = 3, mt = "32px", Se = 4e3, Ne = 356, ht = 14, Ee = typeof window < "u" && typeof document < "u", Ie = /* @__PURE__ */ dt({
  name: "Toaster",
  inheritAttrs: !1,
  __name: "Toaster",
  props: {
    invert: { type: Boolean, default: !1 },
    theme: { default: "light" },
    position: { default: "bottom-right" },
    hotkey: { default: () => ["altKey", "KeyT"] },
    richColors: { type: Boolean, default: !1 },
    expand: { type: Boolean, default: !1 },
    duration: { default: Se },
    gap: { default: ht },
    visibleToasts: { default: Ce },
    closeButton: { type: Boolean, default: !1 },
    toastOptions: { default: () => ({}) },
    className: { default: "" },
    style: { default: () => ({}) },
    offset: { default: mt },
    dir: { default: "auto" },
    containerAriaLabel: { default: "Notifications" },
    pauseWhenPageIsHidden: { type: Boolean, default: !1 },
    cn: {}
  },
  setup(i) {
    function e(...s) {
      return s.filter(Boolean).join(" ");
    }
    function t() {
      if (typeof window > "u" || typeof document > "u")
        return "ltr";
      const s = document.documentElement.getAttribute("dir");
      return s === "auto" || !s ? window.getComputedStyle(document.documentElement).direction : s;
    }
    const n = i, o = Nt(), u = f([]), p = h(() => {
      const s = u.value.filter((r) => r.position).map((r) => r.position);
      return s.length > 0 ? Array.from(new Set([n.position].concat(s))) : [n.position];
    }), y = f([]), c = f(!1), N = f(!1), E = f(
      n.theme !== "system" ? n.theme : typeof window < "u" && window.matchMedia && window.matchMedia("(prefers-color-scheme: dark)").matches ? "dark" : "light"
    );
    h(() => n.position.split("-"));
    const X = h(() => n.cn || e), I = f(null), x = f(null), O = f(!1), et = n.hotkey.join("+").replace(/Key/g, "").replace(/Digit/g, "");
    function L(s) {
      u.value = u.value.filter(({ id: r }) => r !== s.id);
    }
    const G = (s) => {
      var r, v;
      O.value && !((v = (r = s.currentTarget) == null ? void 0 : r.contains) != null && v.call(r, s.relatedTarget)) && (O.value = !1, x.value && (x.value.focus({ preventScroll: !0 }), x.value = null));
    }, at = (s) => {
      s.target instanceof HTMLElement && s.target.dataset.dismissible === "false" || O.value || (O.value = !0, x.value = s.relatedTarget);
    }, ot = (s) => {
      s.target && s.target instanceof HTMLElement && s.target.dataset.dismissible === "false" || (N.value = !1);
    };
    return D((s) => {
      const r = S.subscribe((v) => {
        if (v.dismiss) {
          u.value = u.value.map(
            (B) => B.id === v.id ? { ...B, delete: !0 } : B
          );
          return;
        }
        Et(() => {
          const B = u.value.findIndex(
            ($) => $.id === v.id
          );
          B !== -1 ? u.value.splice(B, 1, v) : u.value = [v, ...u.value];
        });
      });
      s(() => {
        r();
      });
    }), pt(
      () => n.theme,
      (s) => {
        if (s !== "system") {
          E.value = s;
          return;
        }
        s === "system" && (window.matchMedia && window.matchMedia("(prefers-color-scheme: dark)").matches ? E.value = "dark" : E.value = "light"), !(typeof window > "u") && window.matchMedia("(prefers-color-scheme: dark)").addEventListener("change", ({ matches: r }) => {
          r ? E.value = "dark" : E.value = "light";
        });
      }
    ), pt(
      () => I.value,
      () => {
        if (I.value)
          return () => {
            x.value && (x.value.focus({ preventScroll: !0 }), x.value = null, O.value = !1);
          };
      }
    ), D(() => {
      u.value.length <= 1 && (c.value = !1);
    }), D((s) => {
      function r(v) {
        var $, b;
        n.hotkey.every(
          (z) => v[z] || v.code === z
        ) && (c.value = !0, ($ = I.value) == null || $.focus()), v.code === "Escape" && (document.activeElement === I.value || (b = I.value) != null && b.contains(document.activeElement)) && (c.value = !1);
      }
      Ee && (document.addEventListener("keydown", r), s(() => {
        document.removeEventListener("keydown", r);
      }));
    }), (s, r) => (d(), g("section", {
      "aria-label": `${s.containerAriaLabel} ${V(et)}`,
      tabIndex: -1
    }, [
      (d(!0), g(U, null, rt(p.value, (v, B) => {
        var $;
        return d(), g("ol", {
          key: v,
          ref_for: !0,
          ref_key: "listRef",
          ref: I,
          "data-sonner-toaster": "",
          dir: s.dir === "auto" ? t() : s.dir,
          tabIndex: -1,
          "data-theme": s.theme,
          "data-rich-colors": s.richColors,
          "data-y-position": v.split("-")[0],
          "data-x-position": v.split("-")[1],
          style: it(
            {
              "--front-toast-height": `${($ = y.value[0]) == null ? void 0 : $.height}px`,
              "--offset": typeof s.offset == "number" ? `${s.offset}px` : s.offset || mt,
              "--width": `${Ne}px`,
              "--gap": `${ht}px`,
              ...s.style,
              ...V(o).style
            }
          ),
          onBlur: G,
          onFocus: at,
          onMouseenter: r[1] || (r[1] = (b) => c.value = !0),
          onMousemove: r[2] || (r[2] = (b) => c.value = !0),
          onMouseleave: r[3] || (r[3] = () => {
            N.value || (c.value = !1);
          }),
          onPointerdown: ot,
          onPointerup: r[4] || (r[4] = (b) => N.value = !1)
        }, [
          (d(!0), g(U, null, rt(u.value.filter(
            (b) => !b.position && B === 0 || b.position === s.position
          ), (b, z) => {
            var W, _, Y, Z, q, J, Q, tt, R;
            return d(), H(ke, {
              key: b.id,
              index: z,
              toast: b,
              duration: ((W = s.toastOptions) == null ? void 0 : W.duration) ?? s.duration,
              className: (_ = s.toastOptions) == null ? void 0 : _.className,
              descriptionClassName: (Y = s.toastOptions) == null ? void 0 : Y.descriptionClassName,
              invert: s.invert,
              visibleToasts: s.visibleToasts,
              closeButton: ((Z = s.toastOptions) == null ? void 0 : Z.closeButton) ?? s.closeButton,
              interacting: N.value,
              position: s.position,
              style: it((q = s.toastOptions) == null ? void 0 : q.style),
              unstyled: (J = s.toastOptions) == null ? void 0 : J.unstyled,
              classNames: (Q = s.toastOptions) == null ? void 0 : Q.classNames,
              cancelButtonStyle: (tt = s.toastOptions) == null ? void 0 : tt.cancelButtonStyle,
              actionButtonStyle: (R = s.toastOptions) == null ? void 0 : R.actionButtonStyle,
              toasts: u.value,
              expandByDefault: s.expand,
              gap: s.gap,
              expanded: c.value,
              pauseWhenPageIsHidden: s.pauseWhenPageIsHidden,
              cn: X.value,
              heights: y.value,
              "onUpdate:heights": r[0] || (r[0] = (F) => y.value = F),
              onRemoveToast: L
            }, null, 8, ["index", "toast", "duration", "className", "descriptionClassName", "invert", "visibleToasts", "closeButton", "interacting", "position", "style", "unstyled", "classNames", "cancelButtonStyle", "actionButtonStyle", "toasts", "expandByDefault", "gap", "expanded", "pauseWhenPageIsHidden", "cn", "heights"]);
          }), 128))
        ], 44, $e);
      }), 128))
    ], 8, Be));
  }
}), Me = {
  install(i) {
    i.component("Toaster", Ie);
  }
};
export {
  Ie as Toaster,
  Me as default,
  ze as toast
};
